### PR TITLE
Add main screen test

### LINKED ISSUE #1476: Multiple widgets using same GlobalKey
**Describe the bug**

**To Reproduce**

1. Run the app normally and check debug logs

**Expected behavior**
There should be no errors regarding keys and global keys must be used on individual widgets only once

**Actual behavior**
Some widgets are inserted in the widget tree multiple times causing `Multiple widgets used the same GlobalKey` error. This doesn't crash the app, but still is bad and causes problems in testing.

**Screenshots**

Screenshot_20230209_114120

**Additional details**
During testing, I found out that this issue comes from `explore_events.dart` file with this key assignment, along with other pages used in MainScreen Widget

image

The problem is that `navBarItems` is first initialized with all the `BottomNavigationBarItem`, but some of these items are again added to the list in the `renderBottomNavBarPlugins` method.

image

image

image

### Fix

Removing duplicate `BottomNavigationBarItem` addition fixes the problem

I would like to know if this issue is hitting multiple people. @palisadoes @noman2002 @CyberWake @Ayush0Chaudhary

### PR DESCRIPTION
**Pull request by literalEval**
**What kind of change does this PR introduce?**

Adds tests for `MainScreen` class and fixes a bug in the same

**Issue Number:**

Fixes #1476 #1225 

**Did you add tests for your changes?**

Yes, this is a test PR

**Summary**

Added tests for `lib/views/main_screen.dart`, fixed `KeyError` bug and formatted the file a little bit.

**Does this PR introduce a breaking change?**

No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by literalEval** [comment]
Screenshot_20230209_155648

**Comment by palisadoes** [comment]
1. We are going to have delay any potential merge until this PR is complete https://github.com/PalisadoesFoundation/talawa/pull/1474
1. It may cause minor merge conflicts, so be prepared

File: lib/views/main_screen.dart [hunk 1]
```diff
@@ -187,30 +190,51 @@ class _MainScreenState extends State<MainScreen> {
             "class": const ChatPage(
               key: Key('Chats'),
             ),
+          },
+          "Donation": {
+            "class": const ChangeThemeTile(),
```
Comment (noman2002) [inline lib/views/main_screen.dart:195]: We don't need this for now.

File: lib/views/main_screen.dart [hunk 2]
```diff
@@ -165,6 +167,7 @@ class _MainScreenState extends State<MainScreen> {
           "Posts": const Icon(Icons.add_box),
           "Profile": const Icon(Icons.account_circle),
           "Chats": const Icon(Icons.chat_outlined),
+          "Donation": const Icon(Icons.attach_money_outlined),
```
Comment (noman2002) [inline lib/views/main_screen.dart:170]: remove this too

**Review by noman2002 — CHANGES_REQUESTED** [review]

File: lib/views/main_screen.dart (hunk 1 above)
Comment (literalEval) [reply lib/views/main_screen.dart:195]: The problem is with the plugins being added dynamically. If I remove this line, then the execution will never reach to the code with loads plugins (as in the below image). I added this just for the sake of testing purposes. If you have any better ideas on how to test that, please let me know.

image

Comment (noman2002) [reply lib/views/main_screen.dart:195]: @palisadoes Your thoughts on this ?

Comment (palisadoes) [reply lib/views/main_screen.dart:195]: Keep the line and create an issue on how to handle plugins better.

**Comment by palisadoes** [comment]
@noman2002 Please review again

**Review by noman2002 — APPROVED** [review]

File: lib/views/main_screen.dart (hunk 1 above)
Comment (palisadoes) [reply lib/views/main_screen.dart:195]: Yes

**Review by palisadoes — APPROVED** [review]

**Pull request merged**